src/static/riot/analytics/analytics.tag

```diff
@@ -0,0 +1,540 @@
+                    let month = date.getMonth() + 1
+                    let day = date.getDate()
+                    self.end_date = {
+                        year: year,
+                        month: month,
+                        day: day,
+                    }
+
+                    self.update_analytics(self.start_date, self.end_date, self.time_unit, false)
+                },
+            }
+
+            start_calendar_options = _.assign({}, general_calendar_options, start_specific_options)
+            end_calendar_options = _.assign({}, general_calendar_options, end_specific_options)
+
+            $(self.refs.start_calendar).calendar(start_calendar_options);
+            $(self.refs.end_calendar).calendar(end_calendar_options);
+
+            /*---------------------------------------------------------------------
+             Chart Setup
+            ---------------------------------------------------------------------*/
+
+            var ctx_c = self.refs.competition_chart.getContext('2d');
+            self.competitionsChart = new Chart(ctx_c, create_chart_config('# of Competitions'));
+            var ctx_s = self.refs.submission_chart.getContext('2d');
+            self.submissionsChart = new Chart(ctx_s, create_chart_config('# of Submissions'));
+            var ctx_u = self.refs.user_chart.getContext('2d');
+            self.usersChart = new Chart(ctx_u, create_chart_config('# of Users Joined'));
+
+            self.update_analytics(self.start_date, null, self.time_unit, false)
+
+        })
+
+        /*---------------------------------------------------------------------
+         Methods
+        ---------------------------------------------------------------------*/
+
+        function create_chart_config(label) {
+            return {
+                type: 'line',
+                data: {
+                    datasets: [{
+                        label: label,
+                        data: null,
+                        backgroundColor: 'rgba(54, 162, 235, 0.2)',
+                        borderColor:'rgba(54, 162, 235, 1)',
+                        borderWidth: 1,
+                        lineTension: 0,
+                    }],
+                },
+                options: {
+                    scales: {
+                        xAxes: [{
+                            type: 'time',
+                            time: {
+                                unit: 'month'
+                            }
+                        }],
+                        yAxes: [{
+                            ticks: {
+                                beginAtZero: true,
+                                stepSize: 1,
+                            }
+                        }]
+                    }
+                }
+            }
+        }
+
+        function build_chart_data(data, day_resolution, csv_format) {
+            var chart_data = []
+
+            for( let i = 0; i < data.length; i++ ) {
```

Comment:
can simplify this some with lodash
```javascript
let chart_data = _.map(data, data_point => {
    // existing point logic
    return point  // instead of chart_data.push()
}
```

Reply:
What a handy little method.

Reply:
Done.